VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+                IReadOnlyCollection<IWebElement> optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionNames.Count);
+                CollectionAssert.AreEquivalent(_defaultBasicPoll.Options.Select(o => o.Name).ToList(),
+                                               optionNames.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_PopulatedOptions_DisplaysAllOptionDescriptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> optionDescriptions = _driver.FindElements(NgBy.Model("option.Description"));
+
+                Assert.AreEqual(_defaultBasicPoll.Options.Count, optionDescriptions.Count);
+                List<String> expectedDescriptions = new List<string>();
+
+                foreach (Option option in _defaultBasicPoll.Options)
+                {
+                    string truncatedDescription = option.Description
+                                                          .Split(' ')
+                                                          .Aggregate((prev, curr) => (curr.Length + prev.Length >= truncatedTextLimit) ?
+                                                                                     prev : prev + " " + curr);
+
+                    if (truncatedDescription != option.Description)
+                    {
+                        truncatedDescription += "... Show More";
+                    }
+                    expectedDescriptions.Add(truncatedDescription);
+                }
+
+                CollectionAssert.AreEquivalent(expectedDescriptions,
+                                               optionDescriptions.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_VotingOnOption_NavigatesToResultsPage()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
+
+                VoteClearer voterClearer = new VoteClearer(_context);
+                voterClearer.ClearLast();
+
+                Assert.IsTrue(_driver.Url.StartsWith(SiteBaseUri + "Poll/#/Results/" + _defaultBasicPoll.UUID));
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_DefaultPoll_ShowsResultsButton()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _defaultBasicPoll.UUID);
+                IReadOnlyCollection<IWebElement> anchors = _driver.FindElements(By.TagName("a"));
+                IWebElement resultsLink = anchors.FirstOrDefault(a => a.Text == "Go to results");
```

Comment:
Also, this will blow up with a null reference if we ever change the text. FirstOrDefault isn't what we'd want here.